tt_metal/jit_build/data_format.cpp

```diff
@@ -250,14 +264,29 @@ std::vector<DataFormat> get_unpack_dst_formats(DataFormat input_formats[NUM_OPER
         } else if (int_fpu_en) {
             unpack_dst_format.push_back(src_format);
         } else {
-            unpack_dst_format.push_back(get_single_unpack_dst_format(input_formats[i], pack_format, unpack_cond_dst_format));
+            if (!preserve_fp32_precision.empty() && en_unpack_tf32 && preserve_fp32_precision[i] != PreserveFP32Target::Disabled) {
+                TT_FATAL(preserve_fp32_precision[i] == PreserveFP32Target::DEST, "preserve_fp32_precision is only available when unpack target is DEST register");
+                unpack_dst_format.push_back(get_single_unpack_dst_format(input_formats[i], pack_format, DataFormat::Float32));
+            } else {
+                unpack_dst_format.push_back(get_single_unpack_dst_format(input_formats[i], pack_format, unpack_cond_dst_format));
```

Comment:
If `input_formats[i]` is Float32, `pack_format` is Float32, `fp32_dest_acc_en` is true, and `en_unpack_tf32` is false (because there is a non-fp32 input format) then `unpack_dst_format` becomes Float16 at this line. Before this change it was Float32 as intended. Could you check?

Reply:
I see the problem. I've changed the check to only depend on the src data format (no longer depends on en_unpack_tf32 at all. This makes more sense anyways.